backend/src/services/integration-auth/integration-sync-secret.ts

```diff
@@ -462,79 +462,86 @@ const syncSecretsAWSParameterStore = async ({
   ssm.config.update(config);
 
   const metadata = z.record(z.any()).parse(integration.metadata || {});
+  const awsParameterStoreSecretsObj: Record<string, AWS.SSM.Parameter> = {};
+
+  // now fetch all aws parameter store secrets
+  let hasNext = true;
+  let nextToken: string | undefined;
+  while (hasNext) {
```

Comment:
i think we should add some sleep in between each loop so it doesn't rate limit us? what do you think